File: insertion sort/insertion_sort.java
import java.util.Scanner;

class insertion_sort{

    static int n;

    static void sort(int arr[]){
        
        for(int i=1; i<n; i++){
            int j = i - 1;
            int key = arr[i];
            while(j >= 0 && arr[j] > key){
                arr[ j+1 ] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }

    }

    static void display(int arr[]){
        
        for(int i=0; i<n; i++)
            System.out.print(arr[i] + " ");
        System.out.println();

    }
    
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        n = input.nextInt();
        int arr[] = new int[n];

        for(int i=0; i<n; i++)
            arr[i] = input.nextInt();
        
        // insertion_sort obj = insertion_sort();
        sort(arr);
        display(arr);

    }
}

File: insertion sort/insertion_sort.py
def insertion_sort(list):

    for j in range(1, len(list)):
        i = j- 1
        key = list[j]

        while i >= 0 and list[i] > key:
            list[i+1] = list[i]
            i = i - 1
        list[i+1] = key
    
def main():
    list = []
    

    n = int(input("Enter number of elements: "))

    for i in range(0,n):
        list.append(int(input()))

    insertion_sort(list)
    print(list)

if __name__ == "__main__":
    main()